crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,947 @@
-use parcel_core::plugin::PluginContext;
-use parcel_core::plugin::RunTransformContext;
-use parcel_core::plugin::TransformResult;
-use parcel_core::plugin::TransformationInput;
+use std::path::{Path, PathBuf};
+use std::rc::Rc;
+
+use anyhow::{anyhow, Error};
+use indexmap::IndexMap;
+use swc_core::atoms::Atom;
+
 use parcel_core::plugin::TransformerPlugin;
+use parcel_core::plugin::{RunTransformContext, TransformResult, TransformationInput};
+use parcel_core::types::engines::EnvironmentFeature;
+use parcel_core::types::{
+  Asset, AssetFlags, BundleBehavior, Dependency, DependencyFlags, Diagnostic, Environment,
+  EnvironmentContext, FileType, ImportAttribute, JSONObject, Location, OutputFormat, ParcelOptions,
+  Priority, SourceCode, SourceLocation, SourceType, SpecifierType, Symbol, SymbolFlags,
+};
+use parcel_resolver::{ExportsCondition, IncludeNodeModules};
 
+/// This is a rust only `TransformerPlugin` implementation for JS assets that goes through the
+/// default SWC transformer.
 #[derive(Debug)]
-pub struct ParcelTransformerJs {}
+pub struct ParcelJsTransformerPlugin {}
 
-impl ParcelTransformerJs {
-  pub fn new(_ctx: &PluginContext) -> Self {
+impl ParcelJsTransformerPlugin {
+  pub fn new() -> Self {
     Self {}
   }
 }
 
-impl TransformerPlugin for ParcelTransformerJs {
+impl TransformerPlugin for ParcelJsTransformerPlugin {
+  /// This does a lot of equivalent work to `JSTransformer::transform` in
+  /// `packages/transformers/js`
   fn transform(
     &mut self,
-    _context: &mut RunTransformContext,
-    _asset: TransformationInput,
-  ) -> Result<TransformResult, anyhow::Error> {
-    todo!()
+    context: &mut RunTransformContext,
+    input: TransformationInput,
+  ) -> Result<TransformResult, Error> {
+    let file_system = context.file_system();
+    let source_code = input.read_source_code(file_system)?;
+
+    let transformation_result = parcel_js_swc_core::transform(
+      parcel_js_swc_core::Config {
+        filename: input
+          .file_path()
+          .to_str()
+          .ok_or_else(|| anyhow!("Invalid non UTF-8 file-path"))?
+          .to_string(),
+        code: source_code.bytes().to_vec(),
+        source_type: parcel_js_swc_core::SourceType::Module,
+        ..parcel_js_swc_core::Config::default()
+      },
+      None,
+    )?;
+
+    // TODO handle errors properly
+    if let Some(errors) = transformation_result.diagnostics {
+      return Err(anyhow!(format!("{:#?}", errors)));
+    }
+
+    let asset = Asset::new_empty(input.file_path().to_path_buf(), source_code);
+    let config = parcel_js_swc_core::Config::default();
+    let options = ParcelOptions::default();
```

Comment:
This or just `core_path` should be initialised in new

Reply:
I suppose this should be in `context` ;

Reply:
Won't fix in this diff